eng/scripts/TypeSpec-Requirement.ps1

```diff
@@ -8,9 +8,60 @@ param (
 )
 Set-StrictMode -Version 3
 
+Install-Module -Name powershell-yaml -RequiredVersion 0.4.7 -Force -Scope CurrentUser
+
 . $PSScriptRoot/ChangedFiles-Functions.ps1
 . $PSScriptRoot/Logging-Functions.ps1
 
+function Find-Suppressions-Yaml {
+  param (
+    [string]$file
+  )
+
+  $currentDirectory = Get-Item (Split-Path -Path $file)
+
+  while ($currentDirectory) {
+    $suppressionsFile = Join-Path -Path $currentDirectory.FullName -ChildPath "suppressions.yml"
+    if (Test-Path $suppressionsFile) {
+      return $suppressionsFile
+    } else {
+      $currentDirectory = $currentDirectory.Parent
+    }
+  }
+
+  return $null
+}
+
+function Get-Suppression {
+  param (
+    [string]$file
+  )
+
+  $suppressionsFile = Find-Suppressions-Yaml $file
+  if ($suppressionsFile) {
+    $suppressions = Get-Content -Path $suppressionsFile -Raw | ConvertFrom-Yaml
+    foreach ($suppression in $suppressions) {
+      # Disable strict mode to handle tool or path not specified
+      Set-StrictMode -Off
+      $tool = $suppression.tool
```

Comment:
Instead of turning off strict mode you can also just check to see if the property exists similar to:
```
if ($suppression.psobject.Properties.name -contains "tool") { $tool = $suppression.tool }
```

Reply:
@weshaggard: Do you know if this works for hashtables?  I believe $suppression is a hashtable and this only works for actual "objects".  See https://github.com/PowerShell/PowerShell/issues/10875.

I'm not sure if $suppression is always guaranteed to be a hashtable or if this is an implementation detail in powershell-yaml that could change, so I'm not sure if we can rely on it.

Reply:
powershell-yaml docs indicate it's always a hashtable:

```
PS C:\> $obj = ConvertFrom-Yaml $yaml
PS C:\> $obj

Name                           Value
----                           -----
anArray                        {1, 2, 3}
nested                         {array}
hello                          world

PS C:\> $obj.GetType()

IsPublic IsSerial Name                                     BaseType
-------- -------- ----                                     --------
True     True     Hashtable                                System.Object
```

https://github.com/cloudbase/powershell-yaml?tab=readme-ov-file#convertfrom-yaml

Reply:
Yeah I'm unsure if this always a hashtable but if it is you can always call contains key. I believe you can still access it as a psobject to get its properties like I showed as well. Either way I'm fine with.

Reply:
> Yeah I'm unsure if this always a hashtable but if it is you can always call contains key. I believe you can still access it as a psobject to get its properties like I showed as well. Either way I'm fine with.

I don't think so.  You get the properties on the Hashtable object, not the keys.

```
> @{}.psobject.Properties.Name
IsReadOnly
IsFixedSize
IsSynchronized
Keys
Values
SyncRoot
Count
```